### DISCUSSION TITLE
Dropping the mingw-w64-toolchain requirement for makepkg-mingw and building in CI

### DISCUSSION THREAD

**Discussion by lazka** [msys2/MINGW-packages#10506] [Ideas]
I've recently tried to get rid of not explicitly defined dependencies by cleaning up `base-devel` and adding them to `makedepends` in all packages instead, to make `base-devel` smaller and to make the real dependency tree visible to our automation.

Similarly we have those `mingw-w64-XXX-toolchain` groups which we require to be installed when running `makepkg-mingw` where many packages skip to define their `makedepends` if they happen to be pulled in by those groups.

A simple package requiring a C compiler will then pull in `151.34 MiB` of packages instead of the `46.85 MiB` it really needs for building.

Not needing those groups would:

* make CI faster
* help autobuild figure out how to rebuild things without assuming everything depends on the toolchain
* require users to install one thing less before they can start building packages

Thus I propose to remove the requirement for `mingw-w64-XXX-toolchain` groups to be installed when running `makepkg-mingw`  (the groups will stay as is):

* [x] Make all packages makedepend on things they need that are currently provided by the groups
  * [x] Make packages depend on a C compiler -> 7c67d9c46554d10276b2017bcfd0290dfe24a1bf
  * [x] What about pkg-config? -> 0dfda3ffe27c63bf
  * [x] Anything else ? -> We'll see in CI
* [x] Switch CI to only install `cc` -> 0ca9cf2ae0b9740
* [x] Remove the warning about the toolchain missing in `makepkg-mingw` -> https://github.com/msys2/MSYS2-packages/pull/2799
* [x] Update the documentation accordingly -> https://github.com/msys2/msys2.github.io/commit/231b3f8ea1d1952a9dff05cf4fc511c43b10681b
* [x] Don't install anything in CI besides base-devel -> https://github.com/msys2/MINGW-packages/commit/c2c500974ef38b6b77ba1ac7d8410c664b70073f
* [x] Don't install the groups in autobuild -> https://github.com/msys2/msys2-autobuild/commit/6a436ac4e93effddfe49e47819e7387b0ca34504

**Comment by revelator** [comment]
i was wondering if the addition of ${MINGW_PACKAGE_PREFIX}-cc had anything to do with these changes ? since it forces a rebuild of the compiler without other changes.

**Reply by lazka** [reply]
yes, so packages can easily depend on a C compiler. Previously we had clang in clang32/64 provide a virtual "gcc" package, but that confused people and they removed it every time.

**Reply by revelator** [reply]
ah ok i was starting to wonder :)

**Comment by lazka** [comment]
This is all done now.
